### PR TITLE
Keep mailbox in memory

### PR DESCRIPTION
**Pull request by sassmann**
* **What does this PR do?**
add keep_mailbox option

keep_mailbox prevents MailboxViews from being free'd when switching mailboxes. Once a mailbox has been opened it's content is kept in memory. This increases memory consumption, but makes switching between mailboxes lightning fast.
 The default is off to conserve memory.

This is a RFC, feedback is welcome. Please give it a try.

To enable add 
set keep_mailbox=yes
to neomuttrc

### CODE REVIEW

File: mview.c [hunk 1]
```diff
@@ -237,7 +247,7 @@ static void update_tables(struct MailboxView *mv)
   m->changed = false;
   m->msg_flagged = 0;
   padding = mx_msg_padding_size(m);
-  for (i = 0, j = 0; i < m->msg_count; i++)
+  for (i = 0, j = 0; i < m->email_max; i++)
```
Comment (gahr) [inline mview.c:250]: Why this change?

**Comment by gahr** [comment]
Interesting indeed, although I don't quite appreciate the consequences... I'll need to think it through

File: mview.c (hunk 1 above)
Comment (sassmann) [reply mview.c:250]: I've explained that in the commit message, here
https://github.com/neomutt/neomutt/pull/3762/commits/7eef19ede1a19751840b468cc94181a5ede109c6

**Comment by sassmann** [comment]
> Interesting indeed, although I don't quite appreciate the consequences... I'll need to think it through

That's why I made it a config option.

File: mview.c (hunk 1 above)
Comment (gahr) [reply mview.c:250]: Thanks - yep this is also what #3765 is about. I'm not sure we ever want msg_count to exceed the number of emails, so I'm not yet sure what the solution for *that* is going to be.

**Comment by gahr** [comment]
> > Interesting indeed, although I don't quite appreciate the consequences... I'll need to think it through
> 
> That's why I made it a config option.

Sure, but we don't want it to have unintended consequences, even if it's configurable and off by default.

**Comment by vuori** [comment]
Happened to notice this and I think the feature would great to have, since at least the Maildir backend has pretty good code for updating an already-open Mailbox too. But having stared at the relevant bits lately, a comprehensive solution to #3765 aka `msg_count` hermeneutics seems needed first.

Just iterating up to `m->email_max` with a `if (!e)` guard seems surprisingly effective though. IIRC there are only a couple of places where you need to go directly to the end of `m->emails`.

**Comment by flatcap** [comment]
Rebased over `main` (NeoMutt 2023-03-22).

**Comment by flatcap** [comment]
Archived to: https://github.com/neomutt/neomutt-old/tree/keep_mailbox
